### PR TITLE
👁️ Event visibility

### PR DESCRIPTION
**Pull request by JUtupe**

### CODE REVIEW

File: src/main/kotlin/pl/newbies/event/application/EventSchema.kt [hunk 1]
```diff
@@ -33,24 +30,28 @@ class EventSchema(
 ) {
     inner class Query {
         @GraphQLDescription("Get all events paged")
-        fun events(page: Int? = null, size: Int? = null): List<EventResponse> {
+        fun events(
+            page: Int? = null,
+            size: Int? = null,
+            filter: EventFilter? = null,
+            env: DataFetchingEnvironment,
```
Comment (DiD3n) [inline src/main/kotlin/pl/newbies/event/application/EventSchema.kt:37]: 👍

File: src/main/kotlin/pl/newbies/event/domain/service/EventService.kt [hunk 2]
```diff
@@ -1,9 +1,11 @@
 package pl.newbies.event.domain.service
 
 import kotlinx.datetime.Clock
-import org.jetbrains.exposed.sql.SizedCollection
-import org.jetbrains.exposed.sql.and
+import org.jetbrains.exposed.sql.*
```
Comment (DiD3n) [inline src/main/kotlin/pl/newbies/event/domain/service/EventService.kt:4]: whats this 🐜

File: src/main/kotlin/pl/newbies/event/domain/service/EventService.kt [hunk 3]
```diff
@@ -127,4 +148,22 @@ class EventService {
...
+        if (visibilityIn.contains(Event.Visibility.PUBLIC)) {
+            query = query or (Events.visibility eq Event.Visibility.PUBLIC)
+        }
+        if (requesterId != null) {
+            if (visibilityIn.contains(Event.Visibility.INVISIBLE)) {
+                query = query or ((Events.visibility eq Event.Visibility.INVISIBLE) and (Events.author eq requesterId))
+            }
+            if (visibilityIn.contains(Event.Visibility.PRIVATE)) {
+                query = query or ((Events.visibility eq Event.Visibility.PRIVATE) and (Events.author eq requesterId))
+            }
```
Comment (DiD3n) [inline src/main/kotlin/pl/newbies/event/domain/service/EventService.kt:164]: yo, this is for list or single event

File: src/main/kotlin/pl/newbies/event/application/EventController.kt [hunk 4]
```diff
@@ -93,6 +97,23 @@ fun Application.eventRoutes() {
                     call.respond(eventConverter.convert(updatedEvent))
                 }
 
+                put("/{id}/visibility") {
```
Comment (JUtupe) [inline src/main/kotlin/pl/newbies/event/application/EventController.kt:100]: Todo: tests

File: src/main/kotlin/pl/newbies/event/domain/service/EventService.kt (hunk 2 above)
Comment (JUtupe) [reply src/main/kotlin/pl/newbies/event/domain/service/EventService.kt:4]: I'll ignore wildcard imports from now on and configure lint / editor later.

File: src/main/kotlin/pl/newbies/event/application/EventSchema.kt (hunk 1 above)
Comment (JUtupe) [reply src/main/kotlin/pl/newbies/event/application/EventSchema.kt:37]: 👍 👍

File: src/main/kotlin/pl/newbies/event/domain/service/EventService.kt (hunk 3 above)
Comment (JUtupe) [reply src/main/kotlin/pl/newbies/event/domain/service/EventService.kt:164]: I've changed function name, that should clarify this question (It's meant to be used in list)

File: src/main/kotlin/pl/newbies/event/application/EventController.kt (hunk 4 above)
Comment (JUtupe) [reply src/main/kotlin/pl/newbies/event/application/EventController.kt:100]: tests added

**Pull request merged**